src/blueprints/tv2/action-factories/tv2-replay-action-factory.ts

```diff
@@ -0,0 +1,240 @@
+import { PartInterface } from '../../../model/entities/part'
+import { Tv2ActionContentType, Tv2ReplayAction, Tv2ReplayAuxAction } from '../value-objects/tv2-action'
+import { Tv2SourceLayer, Tv2VideoMixerLayer } from '../value-objects/tv2-layers'
+import { PieceLifespan } from '../../../model/enums/piece-lifespan'
+import { TransitionType } from '../../../model/enums/transition-type'
+import { Tv2PieceMetadata } from '../value-objects/tv2-metadata'
+import { Tv2PieceType } from '../enums/tv2-piece-type'
+import { TimelineObject } from '../../../model/entities/timeline-object'
+import {
+  Tv2VideoMixerTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-video-mixer-timeline-object-factory'
+import {
+  Tv2AudioTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-audio-timeline-object-factory'
+import { TimelineEnable } from '../../../model/entities/timeline-enable'
+import { Tv2AudioMode } from '../enums/tv2-audio-mode'
+import { Tv2PieceInterface } from '../entities/tv2-piece-interface'
+
+const EPSIO_REGEX: RegExp = /EPSIO/i
+
+export class Tv2ReplayActionFactory {
+
+  constructor(
+    private readonly videoMixerTimelineObjectFactory: Tv2VideoMixerTimelineObjectFactory,
+    private readonly audioTimelineObjectFactory: Tv2AudioTimelineObjectFactory
+  ) {
+  }
+
+  public createReplayActions(configuration: Tv2BlueprintConfiguration): Action[] {
+    return configuration.studio.SourcesReplay.flatMap(replaySource => {
+      const actions: Action[] = [
+        this.createReplayActionWithVoiceOver(configuration, replaySource),
+        this.createReplayStudioAuxAction(replaySource),
+        this.createReplayVizAuxAction(replaySource)
+      ]
+
+      if (!EPSIO_REGEX.test(replaySource.SourceName)) {
+        actions.push(this.createReplayActionWithoutVoiceOver(configuration, replaySource))
+      }
+
+      return actions
+    })
+  }
+
+  private createReplayActionWithVoiceOver(configuration: Tv2BlueprintConfiguration, source: Tv2SourceMappingWithSound): Tv2ReplayAction {
+    const noWhitespaceName: string = this.removeAllWhitespace(source.SourceName)
+    const partId: string = `${noWhitespaceName}_VO_part_action`
+    const partInterface: PartInterface = this.createPartInterface(partId, `Replay Part ${source.SourceName} VO`)
+    const pieceInterface: Tv2PieceInterface = this.createReplayForSourcePieceInterface(configuration, partId, source, Tv2AudioMode.VOICE_OVER)
+
+    return {
+      id: `insert_${noWhitespaceName}_VO_as_next_part_action`,
+      name: source.SourceName,
+      description: '',
+      type: PartActionType.INSERT_PART_AS_NEXT,
+      data: {
+        partInterface,
+        pieceInterfaces: [
+          pieceInterface
+        ]
+      },
+      metadata: {
+        contentType: Tv2ActionContentType.REPLAY
+      }
+    }
+  }
+
+  private removeAllWhitespace(value: string): string {
+    return value.replace(' ', '')
+  }
```

Comment:
This does not remove all whitespace. It removes all space characters :) tabs, newlines, etc. are kept. If you want to remove all whitespace you can use:
```suggestion
  private removeAllWhitespace(value: string): string {
    return value.replace(/\s+/g, '')
  }
```